lib/util/git/onboarding-branch-cache.ts

```diff
@@ -0,0 +1,42 @@
+import type { ExtractResult } from '../../workers/repository/process/extract-update';
+import { getCache } from '../cache/repository';
+import type { OnboardingBranchCache } from '../cache/repository/types';
+
+export function getCachedOnboardingBranch(
+  branchSha: string,
+  baseBranchSha: string,
+  baseBranchName: string
+): OnboardingBranchCache | null {
+  const cache = getCache();
+  const { onboardingBranch } = cache;
+  if (!onboardingBranch || onboardingBranch?.sha === null) {
```

Comment:
```suggestion
  if (!onboardingBranch || onboardingBranch.sha === null) {
```
can the `sha` be undefined?

Reply:
For the initial run it could be.

Reply:
Shouldn't the object be absent in the initial run?

Reply:
Yes if its undeinfed we return null, incase if the branch is defined but sha is null we return null too. There shouldn't be any case where the branch is defined and sha is null . But all current methods which return branchCommit have return type `string|null` so had to add thiss check due to TS errors.